msal/src/Microsoft.Identity.Client/CacheV2/Schema/Credential.cs

```diff
@@ -0,0 +1,256 @@
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Text;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.CacheV2.Schema
+{
+    internal enum CredentialType
+    {
+        OAuth2AccessToken,
+        OAuth2RefreshToken,
+        OidcIdToken,
+        Other
+    }
+
+    /// <summary>
+    /// This is the object we will serialize (using StorageJson* classes for specific field names) for Credential information.
+    /// Credentials include Access Tokens, Refresh Tokens, etc.
+    /// If you're modifying this object and the related (de)serialization, you're modifying the cache persistence
+    /// model and need to ensure it's compatible and compliant with the other cache implementations.
+    /// </summary>
+    internal class Credential : IEquatable<Credential>
+    {
+        private string _additionalFieldsJson;
+        public string HomeAccountId { get; set; }
+        public string Environment { get; set; }
+        public string Realm { get; set; }
+        public CredentialType CredentialType { get; set; }
+        public string ClientId { get; set; }
+        public string FamilyId { get; set; }
+        public string Target { get; set; }
+        public long CachedAt { get; set; }
+        public long ExpiresOn { get; set; }
+        public long ExtendedExpiresOn { get; set; }
+        public string Secret { get; set; }
+
+        public string AdditionalFieldsJson
+        {
+            get => _additionalFieldsJson;
+            set => _additionalFieldsJson = string.IsNullOrWhiteSpace(value) ? string.Empty : JObject.Parse(value).ToString();
+        }
+
+        public bool Equals(Credential other)
```

Comment:
>other [](start = 38, length = 5)

what is other? can you use a more specific name?

Reply:
This is pretty standard for the Equals comparison method.  It means "the other one you're comparing to"